### PR TITLE
Add badges and some code review details

### PR DESCRIPTION
**Pull request by bobcatfish**
Add to the main README - however note that since we only have PR based
tests right now, if a PR fails, our badges will go red which isn't
really fair to us or the submitter - but we can improve this in #52.

Added a go report card link too - looks like our linting could use some
improvement but we should see how #19 goes (maybe switching out
kubebuilder) before we start following up on all of these.

Looks like:

image

Also adding some details about code reviews after taking with @tejal29
and @aaron-prindle, specifically leaving the reviews up for a bit longer
than we have been to try to give more folks a chance to comment. Also
added a link to some useful Go coding standards.

Fixed a couple typos too.

### CODE REVIEW

**Comment by bobcatfish** [comment]
p.s. 
image


image

File: README.md [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Pipeline CRD
 
+[![Prow unit test results](https://prow.knative.dev/badge.svg?jobs=pull-knative-build-pipeline-unit-tests)](https://prow.knative.dev/?job=pull-knative-build-pipeline-unit-tests)
+[![Prow build results](https://prow.knative.dev/badge.svg?jobs=pull-knative-build-pipeline-build-tests)](https://prow.knative.dev/?job=pull-knative-build-pipeline-build-tests)
+[![Go Report Card](https://goreportcard.com/badge/knative/serving)](https://goreportcard.com/report/knative/serving)
```
Comment (shashwathi) [inline README.md]: I believe this is suppose to be `https://goreportcard.com/report/knative/build-pipeline`.

Comment (bobcatfish) [reply README.md]: whoa thanks for catching that XD @shashwathi

Comment (bobcatfish) [reply README.md]: phew, still green XD and now with way better linting results thanks `knative/serving` :innocent:

**Pull request merged**